src/nl/rubensten/texifyidea/inspections/TexifyRegexInspection.kt

```diff
@@ -114,47 +114,114 @@ abstract class TexifyRegexInspection(
+            val replacements = arrayListOf<String>()
+            val groups = arrayListOf<List<String>>()
+
+            val quickFixName = quickFixName(matcher)
+
+            // For each pattern, just save the replacement location and value
+            while (matcher.find()) {
+                // Pre-checks.
+                if (cancelIf(matcher, file)) {
+                    continue
+                }
+
+                groups.add(groupFetcher(matcher))
+
+                val range = replacementRange(matcher)
+                val replacementContent = replacement(matcher, file)
+
+                // Correct context.
+                val element = file.findElementAt(matcher.start()) ?: continue
+                if (!checkContext(matcher, element)) {
+                    continue
+                }
+
+                replacementRanges.add(range)
+                replacements.add(replacementContent)
             }
 
-            val groups = groupFetcher(matcher)
-            val textRange = highlightRange(matcher)
-            val range = replacementRange(matcher)
-            val error = errorMessage(matcher)
-            val quickFix = quickFixName(matcher)
-            val replacementContent = replacement(matcher, file)
-
-            // Correct context.
-            val element = file.findElementAt(matcher.start()) ?: continue
-            if (!checkContext(matcher, element)) {
-                continue
+            if (replacementRanges.size > 0) {
+
+                // We cannot give one TextRange because there are multiple,
+                // but it does not matter since the user won't see this anyway.
+                val problemDescriptor = manager.createProblemDescriptor(
+                        file,
+                        null as TextRange?,
+                        errorMessage(matcher),
+                        highlight,
+                        isOntheFly,
+                        RegexFixes(
+                                quickFixName,
+                                replacements,
+                                replacementRanges,
+                                groups,
+                                this::applyFixes
+                        )
+                )
+
+                return mutableListOf(problemDescriptor)
+            }
+            else {
+                return mutableListOf()
             }
 
-            descriptors.add(manager.createProblemDescriptor(
-                    file,
-                    textRange,
-                    error,
-                    highlight,
-                    isOntheFly,
-                    RegexFix(
-                            quickFix,
-                            replacementContent,
-                            range,
-                            groups,
-                            this::applyFix
-                    )
-            ))
         }
+        else {
+            val descriptors = SmartList<ProblemDescriptor>()
```

Comment:
```suggestion
            val descriptors = descriptorList()
```
Definitely not wrong, but I think a nice little abstraction that we can slowly incorporate in older inspections.